### ISSUE TITLE
[AVM Module Issue]: Key Vault role assignment not idempotent

### ISSUE THREAD

**Issue by jrlsampension** [Azure/bicep-registry-modules#1884]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/key-vault/vault

### (Optional) Module Version

0.5.1

### Description

When I use RBAC role assignments in this module and try to run the deployment multiple times, I get errors that "The role assignment already exists".

All modules in AVM should be idempotent.

**Comment by AlexanderSehr** [comment]
> ### Check for previous/existing GitHub issues
> * [x]  I have checked for previous/existing GitHub issues
> 
> ### Issue Type?
> Bug
> 
> ### Module Name
> avm/res/key-vault/vault
> 
> ### (Optional) Module Version
> 0.5.1
> 
> ### Description
> When I use RBAC role assignments in this module and try to run the deployment multiple times, I get errors that "The role assignment already exists".
> 
> All modules in AVM should be idempotent.
> 
> ### (Optional) Correlation Id
> _No response_

Hey @jrlsampension,

you're absolutely right, they should - and they usually are. Actually, our tests (usually) run twice in a row to test just that - and I'd argue its very likely the test for KeyVault too.

What I suspect may have happened is the occurence of a very common issue with role assignments in that they're treated as a classic resource even though they are not. But regardless, resource names cannot be updated. This is 'fine' for resources, but in case of role assignments you cannot actually see the name of the role assignment in the backend. This can lead to the very unfortunate behavior that a role assignment may have been applied to the scope you're talking about with name A, while the AVM module tries to apply the same role assignment, but with a different name B. As the name cannot be updated, ARM hence tells you that the 'role assignment already exists' as opposed to just idempotently 'overwrite/ignore' it. 

Long story short, could you please remove the role assignment in question from the resource and run the deployment twice? It 'should' work without any issue 👀

**Comment by ChrisSidebotham** [comment]
@jrlsampension - can you advise if you are still seeing an issue after following @AlexanderSehr steps above?

**Comment by jrlsampension** [comment]
Sorry for my delayed response.

I can confirm that re-running the deployment (and ensuring the deployment name does not change) allows the module to succesfully deploy.

Thank you very much.
